ocaml/parsing/jane_syntax.ml

```diff
@@ -752,6 +749,58 @@ module Layouts = struct
         Desugaring_error.raise ~loc (Unexpected_attribute names)
     in
     lty, attributes
+
+  (*******************************************************)
+  (* Encoding extension constructor *)
+
+  let extension_constructor_of ~loc ~name ~attrs ?info ?docs ext =
+    (* using optional parameters to hook into existing defaulting
+       in Ast_helper.Te.decl, which seems unwise to duplicate *)
+    let module Ast_of = Ast_of (Extension_constructor) (Ext) in
```

Comment:
Won't this recreate the module more times than necessary?  I'd hoist it back to the top level